Update of Enterprise FAQ (#2)
jonico/github_enterprise_deutsch_faq: README.md

* Wie integriert sie Github in SSO Lösungen von Unternehmen (LDAP, Siteminder, NTLM)

GitHub Enterprise unterstützt [SAML](https://help.github.com/en/enterprise/2.19/admin/user-management/using-saml) (Siteminder), [CAS](https://help.github.com/en/enterprise/2.19/admin/user-management/using-cas)[ ](https://help.github.com/en/enterprise/2.19/admin/user-management/using-cas)und [LDAP](https://help.github.com/en/enterprise/2.19/admin/user-management/using-ldap) als Identitätsprovider.  Für eine LDAP-Synch-Schritt- für-Schritt-Anleitung, folgen Sie diesem [Video](https://www.youtube.com/watch?v=Ecjjbe_4md4). Zusätzlich steht eine Synchronisation zwischen LDAP-Zweigen und GitHub-Teammitgliedschaften zur [Verfügung](https://www.youtube.com/watch?v=qxdjGpQjS_g).

## GitHub Actions und GitHub Package Registry

* Was ist GitHub Actions/ GitHub Package Registry?

> [GitHub Actions](https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions) ist GitHubs Platform um CI/CD Aufgaben, wie das builden von code oder das automatisierte Testen von Komponenten, auf GitHub abzubilden. GitHub Actions nutzt hierfür Pipelines, welche durch eine [YAML File](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions) definiert werden. Diese Pipelines bestehen aus sogenannten Actions, welche die einzelnen Schritte der Pipeline darstellen. Eine Action basiert entweder auf JavaScript Code oder einer Dockerfile und liegt in einem eigenen Repository auf GitHub. Es gibt nicht nur Actions von GitHub, sondern auch ein [wachsenden Ökosystem aus Actions](https://github.com/marketplace?type=actions) von anderen Anbietern (etwa Cloud Platformen wie AWS oder Azure oder auch Anbietern wie Atlassian). Eine Actions Pipeline kann auf mehr als [40 Events von GitHub](https://help.github.com/en/actions/reference/events-that-trigger-workflows) reagieren (wie einem Push oder Pull Request) und kann auch extern durch die API getriggered werden. Aus diesen Gründen eignet sich GitHub Actions auch für weitere Automatisierungsvorhaben auf GitHub (etwa die Moderation von Issues oder für Triage Prozesse). Auch kann man [eigene Actions erstellen](https://help.github.com/en/actions/building-actions) und diese wiederum innerhalb der Firma oder mit anderen GitHub Nutzern teilen.

Reviewer: ```suggestion
[GitHub Actions](https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions) ist GitHubs Platform um CI/CD Aufgaben, wie das Bauen von Code oder das automatisierte Testen von Komponenten, auf GitHub abzubilden. GitHub Actions nutzt hierfür Pipelines, welche durch ein [YAML-File](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions) definiert werden. Diese Pipelines bestehen aus sogenannten Actions, welche die einzelnen Schritte der Pipeline darstellen. Eine Action basiert entweder auf JavaScript Code oder einer Dockerfile und liegt in einem eigenen Repository auf GitHub. Es gibt nicht nur Actions von GitHub, sondern auch ein [wachsenden Ökosystem aus Actions](https://github.com/marketplace?type=actions) von anderen Anbietern (etwa Cloud-Platformen wie AWS oder Azure oder Dev-Tool-Herstellern wie Atlassian). Eine Actions Pipeline kann auf mehr als [40 Events von GitHub](https://help.github.com/en/actions/reference/events-that-trigger-workflows) reagieren (wie einem Push oder Pull Request) und kann auch extern durch die API getriggered werden. Aus diesen Gründen eignet sich GitHub Actions auch für weitere Automatisierungsvorhaben auf GitHub (etwa die Moderation von Issues oder für Triage Prozesse). Auch kann man [eigene Actions erstellen](https://help.github.com/en/actions/building-actions) und diese wiederum innerhalb der Firma oder mit anderen GitHub Nutzern teilen.
```

[GitHub Package Registry](https://help.github.com/en/packages/publishing-and-managing-packages/about-github-packages) ist ein in GitHub integrierter Artefakt Storage. Dieser kann aktuell [NPM, Nugget, Ruby Gem, Maven, Gradle und Docker Images](https://help.github.com/en/packages/publishing-and-managing-packages/about-github-packages#supported-clients-and-formats) speichern. Die Package Registry kann sowohl für öffentliche als auch private Projekte verwendet werden und auch direkt aus GitHub Actions als Artefakt Storage verwendet werden. Auch kann Sie in andere Anwendungen eingebunden werden. Vom Funktionsumfang ist die Package Registry aber aktuell nicht mit einem Produkt wie Artifactory vergleichbar.

* Sind GitHub Actions/ GitHub Package Registry für GitHub Enterprise verfügbar?

Aktuell sind beide Produkte für GitHub Enterprise Cloud und Github.com verfügbar. Wir planen später in 2020 die Integration in GitHub Enterprise Server.

> * Kann ich Jenkins Files mit GitHub Actions wiederverwenden?

Reviewer: ```suggestion
* Kann ich Jenkinsfiles mit GitHub Actions wiederverwenden?
```

> Es gibt die Möglichkeit eine [Action](https://github.com/marketplace/actions/jenkinsfile-runner-prepackaged) zu nutzen, welche die Jenkinsfile erkennt und diese mittels eines [Jenkins Single Shot Master](https://github.com/SAP/devops-docker-cx-server/tree/master/jenkinsfile-runner-github-action) innerhalb von GitHub Actions als Docker Container ausführt. Die Jenkinsfile wird dabei innerhalb von Actions ausgeführt. Logs als auch Artefakte können hierbei von Actions abgegriffen und weiterverwendet werden. Dieser Ansatz ermöglicht die Wiederverwendung des bestehenden Jenkins Wissen. Unser Kunde SAP nutzt dies bereits für sein [Cloud SDK](https://blogs.sap.com/2019/09/13/running-sap-cloud-sdk-pipeline-on-github-actions/).

Reviewer: ```suggestion
Es gibt die Möglichkeit eine [Action](https://github.com/marketplace/actions/jenkinsfile-runner-prepackaged) zu nutzen, welche ```Jenkinsfiles``` erkennt und diese mittels eines [Jenkins Single Shot Master](https://github.com/SAP/devops-docker-cx-server/tree/master/jenkinsfile-runner-github-action) innerhalb von GitHub Actions als Docker Container ausführt. Die Jenkins-Pipeline wird dabei innerhalb von Actions ausgeführt. Logs als auch Artefakte können hierbei von Actions abgegriffen und weiterverwendet werden. Dieser Ansatz ermöglicht die Wiederverwendung des bestehenden Jenkins Wissen. SAP nutzt dies bereits für sein [Cloud SDK](https://blogs.sap.com/2019/09/13/running-sap-cloud-sdk-pipeline-on-github-actions/).
```
